services/galley/src/Galley/API/Create.hs

```diff
@@ -118,22 +125,28 @@ createRegularGroupConv zusr zcon (NewConvUnmanaged body) = do
 -- handlers above. Allows both unmanaged and managed conversations.
 createTeamGroupConv :: UserId -> ConnId -> Public.ConvTeamInfo -> Public.NewConv -> Galley ConversationResponse
 createTeamGroupConv zusr zcon tinfo body = do
-  let localUserIds = newConvUsers body
   name <- rangeCheckedMaybe (newConvName body)
+  localDomain <- viewFederationDomain
+  let unqualifiedUserIds = newConvUsers body
+      qualifiedUserIds = newConvQualifiedUsers body
+      allUserIds = map (`Qualified` localDomain) unqualifiedUserIds <> qualifiedUserIds
   let convTeam = cnvTeamId tinfo
   zusrMembership <- Data.teamMember convTeam zusr
+  void $ permissionCheck CreateConversation zusrMembership
+  checkedUsers <- checkedConvSize allUserIds
+  let checkedPartitionedUsers = partitionRemoteOrLocalIds' localDomain <$> checkedUsers
+      (remotes, localUserIds) = fromConvSize checkedPartitionedUsers
   convMemberships <- mapM (Data.teamMember convTeam) localUserIds
   ensureAccessRole (accessRole body) (zip localUserIds convMemberships)
-  void $ permissionCheck CreateConversation zusrMembership
-  otherConvMems <-
+  checkedPartitionedUsers' <-
     if cnvManaged tinfo
       then do
         -- ConvMaxSize MUST be < than hardlimit so the conv size check is enough
         maybeAllMembers <- Data.teamMembersForFanout convTeam
         let otherConvMems = filter (/= zusr) $ map (view userId) $ (maybeAllMembers ^. teamMembers)
-        checkedConvSize otherConvMems
+        checkedLocalUsers <- checkedConvSize otherConvMems
+        pure (fmap ([],) checkedLocalUsers)
```

Comment:
I think I don't understand why an empty list is added here. Maybe add a comment/TODO/futurework?

Reply:
I left a comment for future work, but I believe @pcapriotti and me got to a conclusion that in this branch there can be no remote users.